java/src/main/kotlin/io/spine/validation/java/setonce/SetOnceJavaConstraints.kt

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2024, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.validation.java.setonce
+
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiJavaFile
+import com.intellij.psi.PsiStatement
+import io.spine.protodata.ast.Field
+import io.spine.protodata.java.ClassName
+import io.spine.protodata.java.javaCase
+import io.spine.protodata.java.javaClassName
+import io.spine.protodata.java.render.findClass
+import io.spine.protodata.render.SourceFile
+import io.spine.string.camelCase
+import io.spine.tools.code.Java
+import io.spine.tools.psi.java.Environment.elementFactory
+import io.spine.tools.psi.java.execute
+import io.spine.validation.java.MessageWithFile
+
+/**
+ * Renders Java code to support `(set_once)` option for the given [field].
+ *
+ * The rendered Java constraints are specific to the field type. This class
+ * serves as an abstract base, providing common methods and the skeleton implementation
+ * of [render] method. Inheritors should perform actual rendering in [renderConstraints].
+ *
+ * @property field The field that declared the option.
+ * @property messageWithHeader The message that contains the [field].
```

Comment:
Please rename to `declaredIn`. Assuming the usage of this property it would be shorter and look a bit more readable.